driver.go

```diff
@@ -47,10 +47,18 @@ import (
 // Driver is exported to allow it to be used directly.
 type Driver struct{}
 
-// Open a Connection to the server.
-// See https://github.com/apache/calcite-avatica-go#dsn for more information
-// on how the DSN is formatted.
-func (a *Driver) Open(dsn string) (driver.Conn, error) {
+// Connector implements the driver.Connector interface
+type Connector struct {
+	additionalInfos   map[string]string
+	additionalHeaders map[string]string
+}
+
+// NewConnector creates a new connector
+func NewConnector(additionalInfos, additionalHeaders map[string]string) driver.Connector {
+	return &Connector{additionalInfos, additionalHeaders}
+}
+
+func (c *Connector) Connect(context.Context) (driver.Conn, error) {
 
 	config, err := ParseDSN(dsn)
```

Comment:
How does this code get access to the `dsn` variable?

Reply:
was a bug since it referenced a global variable. fixed.